src/wallet/walletdb.h

```diff
@@ -294,4 +294,7 @@ class WalletBatch
 //! Compacts BDB state so that wallet.dat is self-contained (if there are changes)
 void MaybeCompactWalletDB();
 
+//! Unserialize a give Key-Value pair and load it into the wallet
+bool ReadKeyValue(CWallet* pwallet, CDataStream& ssKey, CDataStream& ssValue, std::string& strType, std::string& strErr);
```

Comment:
in commit 6d05dff02b:

wallet is a reference to an object (like all other params in this function), so it should probably be passed as one.

Not sure about the lock annotation, but it is a bit confusing to lock cs_wallet, and then call a function that locks it again immediately after

```suggestion
bool ReadKeyValue(CWallet& wallet, CDataStream& ssKey, CDataStream& ssValue, std::string& strType, std::string& strErr) EXCLUSIVE_LOCKS_REQUIRED(wallet.cs_wallet);
```

Reply:
The annotation causes a compile error as `CWallet` is not yet defined.

Reply:
Fair enough